demos/TicTacToe/Classes/tic_tac_toe_layer.cc

```diff
@@ -449,101 +425,155 @@ TicTacToeLayer::TicTacToeLayer(string game_uuid,
       ->addEventListenerWithSceneGraphPriority(touch_listener, board_sprite_);
 
   this->addChild(board_sprite_);
-
-  // Schedule the update method for this scene.
-  this->scheduleUpdate();
 }
 
-// Called automatically every frame. The update is scheduled in constructor.
-void TicTacToeLayer::update(float /*delta*/) {
-  // Replace the scene with MainMenuScene if the initialization fails.
-  if (initialization_failed_ == true) {
-    Director::getInstance()->popScene();
-  }
+// If the join_game_uid_ is present, initialize game variables, otherwise
+// alter the game variables to signify a user joined. Additionally sets the
+// player_index_ and total_players based on joining or creating a game.
+void TicTacToeLayer::InitializePlayerData() {
+  if (join_game_uid_.empty()) {
+    join_game_uid_ = GenerateUid(4);
+    ref_ = database_->GetReference("game_data").Child(join_game_uid_);
+    future_create_game_ = ref_.Child("total_players").SetValue(1);
```

Comment:
@alexames should this create a future, then wait, create a future, then wait, etc.  Instead of creating all the futures, then wait on all the futures?

Reply:
There's no reason not to kick of multiple long running operations at once. Ideally we wouldn't block on them at all, but there's a separate issue tracking their removal so he can change that later